src/linker/Linker/LinkContext.cs

```diff
@@ -468,10 +468,84 @@ public bool IsOptimizationEnabled (CodeOptimizations optimization, AssemblyDefin
 
 		public void LogMessage (MessageContainer message)
 		{
-			if (LogMessages && message != MessageContainer.Empty)
+			if (LogMessages)
 				Logger?.LogMessage (message);
 		}
 
+		public void LogMessage (string message, bool isDiagnostic = false)
```

Comment:
I'd go with `LogDiagnostic` instead of extra argument which you have to type all the time to know what it's for